pkg/template/core/starlark_struct.go

```diff
@@ -49,14 +49,15 @@ func (s *StarlarkStruct) AttrNames() []string {
 }
 
 func (s *StarlarkStruct) Get(key starlark.Value) (val starlark.Value, found bool, err error) {
-	if attr, ok := NewStarlarkValue(key).AsGoValue().(string); ok {
-		val, found := s.data.Get(attr)
-		if found {
-			return val.(starlark.Value), true, nil
-		}
-		return starlark.None, false, nil
+	attrName, err := NewStarlarkValue(key).AsString()
+	if err != nil {
+		return starlark.None, false, fmt.Errorf("expected key `%s` to be a string but is a %s", key, key.Type())
```

Comment:
do you think we still need to have this error message here? (assumption here is that https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/template/core/starlark_value.go#L37 should be good enough, if not can we improve it there?) if we do want to keep this message here, then we should start it with capital letter (to be consistent with other error messages) and use single quotes instead of ticks (ticks arent used in any carvel tools).